src/cli/commands/execute/compile.rs

```diff
@@ -19,7 +21,7 @@ const JSON_FILE_EXTENTION: &str = "json";
 /// let mut program = PathBuf::from("path_to_your_program");
 /// let compiled_program_path = compile(&program);
 /// ```
-pub fn compile(program: &PathBuf) -> PathBuf {
+pub fn compile(program: &PathBuf) -> Result<PathBuf, String> {
```

Comment:
You should return an error when the compilation failed.
Here I don't see any error being returned.

You can keep most of the `unwrap` and `expect` for now, but we have to be sure `Command::new` was successfull

Reply:
oh this is a hackish solution because the signature of the exec have this following form fn exec(&self) -> Result<TestOutput, String>, do I still need to change the type?

Reply:
The compilation can fail, and we want an error to be returned when it fails.
So you have to check that the compilation process was actually successful.
You can do so using: https://doc.rust-lang.org/std/process/struct.Command.html#method.status

Reply:
done!